native/src/routes/SearchModal.tsx

```diff
@@ -59,29 +47,10 @@ const SearchModal = ({
   const [query, setQuery] = useState<string>(initialSearchText)
   const resourceCache = useResourceCache({ cityCode, languageCode })
 
-  const searchResults = useMemo(() => searchCategories(categories, query), [categories, query])
+  const minisearch = useMiniSearch(allPossibleResults)
 
-  const onItemPress = (category: CategoryModel): void => {
-    const routeInformation: CategoriesRouteInformationType = {
-      route: CATEGORIES_ROUTE,
-      cityContentPath: category.path,
-      cityCode,
-      languageCode,
-    }
-    sendTrackingSignal({
-      signal: {
-        name: SEARCH_FINISHED_SIGNAL_NAME,
-        query,
-        url: urlFromRouteInformation(routeInformation),
-      },
-    })
-    navigateTo({
-      route: CATEGORIES_ROUTE,
-      cityCode,
-      languageCode,
-      cityContentPath: category.path,
-    })
-  }
+  // Minisearch doesn't add the returned storeFields (e.g. title or path) to its typing
+  const searchResults = minisearch.search(query) as unknown as SearchResult[]
```

Comment:
Okay, I investigated this a little more. On web this just does not throw an error because we are not requiring the type to be (our) SearchResult. Instead, we are just destructuring it straight away and the return value of `.search()` includes `    [key: string]: any;`, therefore all of our fields are just of type `any`.

Since that (as well as just typecasting) is a little unsafe as we won't be notified when this is fixed or we change used fields) I would suggest the following: Include the actual `search` process in the hook and use as follows:
``` ts
const useSearch = (allPossibleResults: SearchResult[], query: string): SearchResult[] => {
  const minisearch: MiniSearch<SearchResult> = useMemo(() => {
    ...
  }, [allPossibleResults])

  // @ts-expect-error minisearch doesn't add the returned storeFields (e.g. title or path) to its typing
  return query.length === 0 ? allPossibleResults : minisearch.search(query)
}
```

The cool thing about that is that how we are doing the search is just an implementation detail, we only have to handle it once and `ts-expect-error` notifies us if this is fixed by the library.

Optional: Rename the hook to `useSearch` as using minisearch is just an implementation detail.

Then you could do this here and remove the conditional further below:
```suggestion
  const searchResults = useSearch(allPossibleResults, query)
```

Reply:
Very cool, thanks for looking further into it, I like it!